primitives/genesis-builder/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! Substrate genesis config builder
+//!
+//! This Runtime API allows to construct `GenesisConfig`, in particular:
+//! - serialize the runtime default `GenesisConfig` struct into json format,
+//! - put the GenesisConfig struct into the storage. Internally this operation calls
+//!   `GenesisBuild::build` function
+//! for all runtime pallets, which is typically provided by pallet's author.
+//! - deserialize the GenesisConfig from given json blob and put GenesisConfig into the state
+//!   storage. Allows to build
+//! customized configuration.
+//!
+//! Providing externalities with empty storage and putting GenesisConfig into storage allows to
+//! catch and build the raw storage of `GenesisConfig` which is the foundation for genesis block.
+
+sp_api::decl_runtime_apis! {
+	/// API to interact with GenesisConfig for the runtime
+	pub trait GenesisBuilder {
+		/// Get the default `GenesisConfig` as a JSON blob.
+		///
+		/// This function instantiates the default `GenesisConfig` struct for the runtime and serializes it into
+		/// a JSON blob.
+		///
+		/// # Returns
```

Comment:
While I favor having documented arguments and return values explicitly (as you've done here), looks like the general repo doesn't follow any guideline.
In general (and this is a personal opinion as well 😃) I really dislike mixing opinionated styles within the same project.

Probably we should definitely open an issue about improving the doc containing the code guidelines that are not explicitly driven by `rustfmt` (docs/CODING_STYLE.md)

I already proposed it once and probably I'm going to open it soon 😃

But then who is going to decide what is the best for the whole project? I mean this kind of things needs some form of collective consensus I suppose

Reply:
Removed in: https://github.com/paritytech/substrate/pull/14131/commits/254776103e172630efd200cee13aecbc2a032e70